jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/PluginServiceFactoryNestedContextTest.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.util;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class PluginServiceFactoryNestedContextTest {
+  private PluginServiceFactory<TestContext> pluginServiceFactory;
+
+  @BeforeEach
+  public void setup() {
+    pluginServiceFactory = new PluginServiceFactory<>(new DefaultTestContext());
+  }
+
+  @Test
+  void createServiceObjects_withConstructorParameterSubtype_shouldReturn() {
```

Comment:
maybe rename this method: from what I understand:
passing a sub interface, should match the Service implementation, create and return an instance of the Service implementation. right ?